### PR TITLE
Add possibility of config file to CLI

### LINKED ISSUE #485: Add log4j file to cslang-cli.zip
Add "log4j.properties" (under cloudslang-cli - >resources ) to cslang-cli.zip so the user would be able  to configure the file:
1. add it as a task to the build_cli_flow.sl
2. configure spring shell to the new location of log4j file

# suggestion for solution:
*option 1:* create "resources" folder in the zip which will contain log4j file. spring shell supposed to take log4j file from "resources" folder by default.
*option 2:* explicitly mentioning the location of log4j by adding it to "application.properties" file:
logging.config= # Location of the logging configuration file. For instance `classpath:logback.xml` for Logback

(we can check how this is implemented in other applications- maybe in OO and decide)

links:
http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html

http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-logging

### PR DESCRIPTION
**Pull request by Bonczidai**
fixes #485 
fixes partially #423  

Signed-off-by: Bonczidai Levente <levente.bonczidai@hpe.com>

### CODE REVIEW

File: cloudslang-cli/src/main/java/io/cloudslang/lang/cli/SlangBootstrap.java [hunk 1]
```diff
@@ -21,9 +28,50 @@
...
         System.out.println("Loading..");
         Bootstrap.main(args);
     }
 
+    private static void loadUserProperties() throws IOException {
+        String appHome = System.getProperty("app.home");
+        String propertyFilePath = appHome + File.separator + USER_CONFIG_FILEPATH;
+        File propertyFile = new File(propertyFilePath);
+        Properties rawProperties = new Properties();
+        if (propertyFile.exists()) {
```
Comment (gituser173) [inline cloudslang-cli/src/main/java/io/cloudslang/lang/cli/SlangBootstrap.java]: you could use isFile() here, checks if exists and if it is a file

File: cloudslang-cli/src/main/java/io/cloudslang/lang/cli/SlangBootstrap.java [hunk 2]
```diff
@@ -21,9 +28,50 @@
...
+            try (InputStream propertiesStream = new FileInputStream(propertyFilePath)) {
+                rawProperties.load(propertiesStream);
+            }
+        }
+        Enumeration<?> e = rawProperties.propertyNames();
```
Comment (gituser173) [inline cloudslang-cli/src/main/java/io/cloudslang/lang/cli/SlangBootstrap.java]: maybe use the entrySet() here to get both name and value

File: cloudslang-cli/src/main/java/io/cloudslang/lang/cli/SlangBootstrap.java [hunk 3]
```diff
@@ -21,9 +28,50 @@
...
+            System.setProperty(key, value);
+        }
+    }
+
+    private static String substitutePropertyReferences(String value) {
+        Matcher mather = SUBSTITUTION_PATTERN.matcher(value);
+        Set<String> variableNames = new HashSet<>();
+        while (mather.find()) {
+            variableNames.add(mather.group(1));
+        }
```
Comment (gituser173) [inline cloudslang-cli/src/main/java/io/cloudslang/lang/cli/SlangBootstrap.java]: maybe split this method into two methods at this line, the above part "findSystemPropertiesNames", the below part "replaceSystemPropertiesNamesWithValues" or something similar. WDYT?

File: cloudslang-cli/src/main/java/io/cloudslang/lang/cli/SlangBootstrap.java [hunk 4]
```diff
@@ -21,9 +28,50 @@
  */
 public class SlangBootstrap {
 
+    private static final String USER_CONFIG_DIR = "configuration";
+    private static final String USER_CONFIG_FILENAME = "cslang.properties";
+    private static final String USER_CONFIG_FILEPATH = USER_CONFIG_DIR + File.separator + USER_CONFIG_FILENAME;
+    private static final String SUBSTITUTION_REGEX = "\\$\\{([^${}]+)\\}"; // ${VAR}
```
Comment (gituser173) [inline cloudslang-cli/src/main/java/io/cloudslang/lang/cli/SlangBootstrap.java]: // ${SystemPropertyName} instead of ${VAR} would be more concise maybe?

**Comment by Bonczidai** [comment]
please note note to delete feature branches - like this one -  in this sprint, they will also need to be merged in v1.0

**Pull request merged**